web/src/settings_users.ts

```diff
@@ -190,29 +190,51 @@ function role_selected_handler(
     widget.render();
 }
 
-function get_roles(): {
-    unique_id: number;
-    name: string;
-}[] {
+function count_users_by_role(user_ids: number[]): Record<number, number> {
+    const role_counts: Record<number, number> = {};
+
+    for (const user_id of user_ids) {
+        const user = people.get_by_user_id(user_id);
+        const role_code = user.role;
+
+        role_counts[role_code] = (role_counts[role_code] ?? 0) + 1;
+    }
+
+    return role_counts;
+}
+
+function get_roles_with_counts(user_ids: number[]): {unique_id: number; name: string}[] {
+    const role_counts = count_users_by_role(user_ids);
     return [
-        {unique_id: 0, name: $t({defaultMessage: "All roles"})},
+        {
+            unique_id: 0,
+            name: $t({defaultMessage: "All roles ({count})"}, {count: user_ids.length}),
+        },
         ...Object.values(settings_config.user_role_values)
             .map((user_role_value) => ({
                 unique_id: user_role_value.code,
-                name: user_role_value.description,
+                name: $t(
+                    // This translation is a noop except for RTL languages
+                    {defaultMessage: "{description} ({count})"},
+                    {
+                        description: user_role_value.description,
+                        count: role_counts[user_role_value.code] ?? 0,
+                    },
+                ),
             }))
             .reverse(),
     ];
 }
 
 function create_role_filter_dropdown(
     $events_container: JQuery,
-    section: UserSettingsSection,
+    section: {dropdown_widget_name: string; filters: {role_code: number}},
```

Comment:
Why did you make this change? Looks like a mistake -- perhaps you mis-resolved a typescript type error?

The previous type was correct.